spec/features/admin/invoice_print_spec.rb

```diff
@@ -120,6 +123,237 @@
 
   it_behaves_like "Check display on each invoice: legacy and alternative", false
   it_behaves_like "Check display on each invoice: legacy and alternative", true
+
+  describe "an order with taxes" do
+    let(:user1) { create(:user, enterprises: [distributor]) }
+    let!(:zone) { create(:zone_with_member) }
+    let(:address) { create(:address) }
+
+    context "included" do
+      let(:shipping_tax_rate_included) {
+        create(:tax_rate, amount: 0.20, included_in_price: true, zone: zone)
+      }
+      let(:shipping_tax_category) { create(:tax_category, tax_rates: [shipping_tax_rate_included]) }
+      let!(:shipping_method) {
+        create(:shipping_method_with, :expensive_name, distributors: [distributor],
+                                                       tax_category: shipping_tax_category)
+      }
+      let(:enterprise_fee) {
+        create(:enterprise_fee, enterprise: user1.enterprises.first,
+                                tax_category: product2.tax_category,
+                                calculator: Calculator::FlatRate.new(preferred_amount: 120.0))
+      }
+      let(:order_cycle) {
+        create(:simple_order_cycle, coordinator: distributor,
+                                    coordinator_fees: [enterprise_fee], distributors: [distributor],
+                                    variants: [product1.variants.first, product2.variants.first])
+      }
+
+      let(:order1) {
+        create(:order, order_cycle: order_cycle, distributor: user1.enterprises.first,
+                       ship_address: address, bill_address: address)
+      }
+      let(:product1) {
+        create(:taxed_product, zone: zone, price: 12.54, tax_rate_amount: 0,
+                               included_in_price: true)
+      }
+      let(:product2) {
+        create(:taxed_product, zone: zone, price: 500.15, tax_rate_amount: 0.2,
+                               included_in_price: true)
+      }
+
+      let!(:line_item1) {
+        create(:line_item, variant: product1.variants.first, price: 12.54, quantity: 1,
+                           order: order1)
+      }
+      let!(:line_item2) {
+        create(:line_item, variant: product2.variants.first, price: 500.15, quantity: 3,
+                           order: order1)
+      }
+
+      before do
+        order1.reload
+        break unless order1.next! until order1.delivery?
```

Comment:
I'm not entirely sure how to address this one. Any suggestions?

Reply:
Could this be remplaced by 
```ruby
        until order1.delivery?
          break unless order1.next!
        end
```

But I think I do prefer the version:

```ruby
while !order1.delivery?
  break if !order1.next!
end
```

Reply:
> But I think I do prefer the version:

Thanks JB. Commited and squashed this second suggestion:+1: